.github/workflows/prerelease-tests.yml

```diff
@@ -0,0 +1,164 @@
+name: Pre-release tests
+
+on:
+    pull_request:
+        branches: 
+            - main
+
+jobs:
+    test-jupyterlab-demos:
+        runs-on: ubuntu-20.04
+        timeout-minutes: 60
+        strategy:
+          matrix:
+            python-version: ['3.7', '3.8', '3.10']
+          fail-fast: false
+    
+        steps:
+        - name: Cancel Previous Runs
+          uses: styfle/cancel-workflow-action@0.7.0
+          with:
+            access_token: ${{ github.token }}
+        - uses: actions/checkout@v2
+        - name: Set up Python ${{ matrix.python-version }}
+          uses: actions/setup-python@v2
+          with:
+            python-version: ${{ matrix.python-version }}
+            cache: pip
+            cache-dependency-path: |
+              mitosheet/setup.py
+              tests/requirements.txt
+        - uses: actions/setup-node@v3
+          with:
+            node-version: 16
+            cache: 'npm'
+            cache-dependency-path: mitosheet/package-lock.json
+        - name: Install dependencies
+          run: |
+            cd tests
+            python3 -m venv venv
+            source venv/bin/activate
+            pip install mitosheet
+            pip install --upgrade -i https://test.pypi.org/simple/ mitosheet
+            # Install necessary node packages
+            npm install
+            npx playwright install chromium webkit firefox chrome
+
+        - name: Start a server and run tests
+          run: |
+            cd tests
+            source venv/bin/activate
+            jupyter lab --config jupyter_server_test_config.py &
+            npm run test:jupyterlab
```

Comment:
This is just running jupyter lab tests, no?

Reply:
Yes, but on the test pypi version.

Reply:
Also: I wanted to add a more in-depth jupyterlab demo test but ran out of time!